### PR TITLE
@CONST belongs on class in TypeScript, but emits on ctor in dart

### LINKED ISSUE #140: @CONST on class emits const in wrong place
reported by @vsavkin 

Dart doesn't allow `const class`. I'll add a repro in the e2e test.

### PR DESCRIPTION
**Pull request by alexeagle**
fixes #140

### CODE REVIEW

**Comment by alexeagle** [comment]
hmm, hold on this one, need to fix something...

File: lib/call.ts [hunk 1]
```diff
@@ -98,13 +98,13 @@ class CallTranspiler extends base.TranspilerStep {
     var errorAssignmentsSuper = 'const constructors can only contain assignments and super calls';
     var errorThisAssignment = 'assignments in const constructors must assign into this.';
 
-    var isConstCtor = this.hasAnnotation(ctor.decorators, 'CONST');
+    var isConstClass = this.hasAnnotation(ctor.parent.decorators, 'CONST');
```
Comment (mprobst) [inline lib/call.ts]: just do || with ctor.decorators?

File: lib/declaration.ts [hunk 2]
```diff
@@ -92,7 +95,7 @@ class DeclarationTranspiler extends base.TranspilerStep {
       case ts.SyntaxKind.PropertyDeclaration:
         var propertyDecl = <ts.PropertyDeclaration>node;
         this.visitDeclarationMetadata(propertyDecl);
-        var hasConstCtor = this.hasConstCtor(<ClassLike>propertyDecl.parent);
+        var hasConstCtor = this.hasAnnotation((<ClassLike>propertyDecl.parent).decorators, 'CONST');
```
Comment (mprobst) [inline lib/declaration.ts]: rename to hasConstClass? Also, at this point it might be smarter to extract a method.

File: lib/declaration.ts [hunk 3]
```diff
@@ -298,6 +301,14 @@ class DeclarationTranspiler extends base.TranspilerStep {
     }
     this.emit('{');
     this.visitEachIfPresent(decl.members);
+
+    // Generate a constructor to host the const modifier, if needed
+    if (this.hasAnnotation(decl.decorators, 'CONST') &&
```
Comment (mprobst) [inline lib/declaration.ts]: use `isConstClass` from above?

File: lib/call.ts (hunk 1 above)
Comment (alexeagle) [inline lib/call.ts]: done

File: lib/declaration.ts (hunk 2 above)
Comment (alexeagle) [inline lib/declaration.ts]: done

**Comment by alexeagle** [comment]
ready to merge

File: lib/declaration.ts [hunk 4]
```diff
@@ -377,11 +381,12 @@ class DeclarationTranspiler extends base.TranspilerStep {
     }
   }
 
-  private hasConstCtor(decl: ClassLike) {
-    return decl.members.some((m) => {
-      if (m.kind !== ts.SyntaxKind.Constructor) return false;
-      return this.hasAnnotation(m.decorators, 'CONST');
-    });
+  private isConst(decl: ClassLike) {
```
Comment (mprobst) [inline lib/declaration.ts]: move into base so this can be shared?

**Comment by mprobst** [comment]
LGTM

File: lib/declaration.ts (hunk 4 above)
Comment (alexeagle) [inline lib/declaration.ts]: yes, and also, I like VSCode.

**Pull request merged**